workspaces/tech-insights/plugins/tech-insights-backend-module-jsonfc/src/service/JsonRulesEngineFactChecker.ts

```diff
@@ -347,13 +348,17 @@ export class JsonRulesEngineFactChecker
       .filter(factContainer =>
         techInsightCheck.factIds.includes(factContainer.id),
       )
-      .reduce(
-        (acc, factContainer) => ({
+      .reduce((acc, factContainer) => {
+        const factWithTimestamp = {
+          ...factContainer.facts,
+          timestamp: factContainer.timestamp,
+        };
+
+        return {
           ...acc,
-          ...pick(factContainer.facts, individualFacts),
-        }),
-        {},
-      );
+          ...pick(factWithTimestamp, individualFacts),
```

Comment:
Does this actually eventually return the timestamp?

Reply:
@Xantier - late response. we hacked something together and needed to move on. It would still be nice to have the timestamp included in the `FactContainer` for ease of access and this should solve for that.